### PR TITLE
Report climate assessment variables

### PR DESCRIPTION
**Pull request by tonnrueter**
- Function  `reportClimate` extracts variables `pm_globalMeanTemperature` & `p15_forc_magicc` from `fulldata.gdx`
- Adapted `convGDX2MIF.R` to include climate assessment variables

### CODE REVIEW

File: R/reportClimate.R [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#' @examples
+#' \dontrun{
+#' reportClimate(gdx)
+#' }
+#'
+#' @export
+#' @importFrom gdx readGDX
+#' @importFrom magclass setNames mbind
+reportClimate <- function(gdx) {
+  tryCatch(
```
Comment (fbenke-pik) [inline R/reportClimate.R:17]: tryCatch can most likely be avoided here. `readGDX` has a `react` parameter  that can be set to `silent` to return NULL

File: R/reportClimate.R [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    # Climate assessment is turned of per default. Try reading climate asessment data from GDX, but fail
+    # silently if none are available.
+    {
+      variables <- c(
+        pm_globalMeanTemperature = "Temperature|Global Mean (K)",
+        p15_forc_magicc = "Forcing (W/m2)"
+      )
+      return(mbind(lapply(names(variables), function(gamsVariable) {
```
Comment (fbenke-pik) [inline R/reportClimate.R:25]: I am not a fan of lapply here. Especially, if you are looping only over two variables. These lines are hard to read, understand and debug. You should simply write two readGDX statements for the sake of clarity.

File: R/convGDX2MIF.R [hunk 3]
```diff
@@ -126,6 +126,30 @@ convGDX2MIF <- function(gdx, gdx_ref = NULL, file = NULL, scenario = "default",
     message("function reportSDPVariables does not work and is skipped")
   }
 
+  # Report climate assessment variables & merge with output
+  message("running reportClimate...")
+  climateVars <- reportClimate(gdx) # Returns NULL if no climate assessment variables are available
+  if (!is.null(climateVars)) {
```
Comment (fbenke-pik) [inline R/convGDX2MIF.R:132]: All this should be moved to `reportClimate`. convGDX2MIF is the hub function and should only have two lines of code, sth along the lines of

```
message("running reportClimate...")
output <- mbind(output,reportClimate(gdx, ...))
```

File: R/reportClimate.R (hunk 1 above)
Comment (tonnrueter) [reply R/reportClimate.R]: Done

File: R/convGDX2MIF.R (hunk 3 above)
Comment (tonnrueter) [reply R/convGDX2MIF.R]: Moved to `reportClimate.R`

**Review by fbenke-pik — APPROVED** [review]

**Pull request merged**